File: VB/SchedulingDemo.Wpf/ViewModels/OutlookInspired/CalendarDetailPrintReportViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Xpf.Scheduling
Imports DevExpress.XtraReports

Namespace SchedulingDemo.ViewModels
    Public Class CalendarDetailPrintReportViewModel
        Public Shared Function Create() As CalendarDetailPrintReportViewModel
            Return ViewModelSource.Create(Function() New CalendarDetailPrintReportViewModel())
        End Function

        Protected Sub New()
            PrintStyle = CalendarDetailPrintStyle.DailyStyle
        End Sub

        Public Overridable Property PrintStyle() As CalendarDetailPrintStyle
        Public Overridable Property Report() As IReport

        Public Sub UpdateReport(ByVal scheduler As SchedulerControl)
            Report = SchedulerReportFactory.Create(PrintStyle, scheduler)
            Report.PrintingSystemBase.ClearContent()
            Report.CreateDocument(True)
        End Sub
    End Class
End Namespace

File: CS/SchedulingDemo.Wpf/ViewModels/OutlookInspired/CalendarDetailPrintReportViewModel.cs
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.Scheduling;
using DevExpress.XtraReports;

namespace SchedulingDemo.ViewModels {
    public class CalendarDetailPrintReportViewModel {
        public static CalendarDetailPrintReportViewModel Create() {
            return ViewModelSource.Create(() => new CalendarDetailPrintReportViewModel());
        }

        protected CalendarDetailPrintReportViewModel() {
            PrintStyle = CalendarDetailPrintStyle.DailyStyle;
        }

        public virtual CalendarDetailPrintStyle PrintStyle { get; set; }
        public virtual IReport Report { get; set; }

        public void UpdateReport(SchedulerControl scheduler) {
            Report = SchedulerReportFactory.Create(PrintStyle, scheduler);
            Report.PrintingSystemBase.ClearContent();
            Report.CreateDocument(true);
        }
    }
}
